crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -384,16 +399,34 @@ impl RealRibosome {
         &self,
         zome_name: &ZomeName,
     ) -> RibosomeResult<Arc<ModuleWithStore>> {
-        Ok(self.serialized_module_cache.write().get(
-            self.wasm_cache_key(zome_name)?,
-            &self.dna_file.get_wasm_for_zome(zome_name)?.code(),
-        )?)
+        use holochain_wasmer_host::module::PlruCache;
+
+        let cache_key = self.get_module_cache_key(zome_name)?;
+
+        let mut module_cache = self.module_cache.write();
+        if let Some(module_with_store) = module_cache.get_item(&cache_key) {
+            tracing::error!("found a module in the module cache");
+            return Ok(module_with_store);
+        }
```

Comment:
This is the reinstated deserialized module cache. Check first if there's a module in the cache, otherwise deserialize it from the serialized module.